src/ServerRequestFactoryInterface.php

```diff
@@ -12,20 +12,9 @@
      *
      * @param string $method
      * @param UriInterface|string $uri
+     * @param array $server optional server params ($_SERVER or similar structure)
      *
      * @return ServerRequestInterface
      */
-    public function createServerRequest($method, $uri);
-
-    /**
-     * Create a new server request from server variables.
-     *
-     * @param array $server Typically $_SERVER or similar structure.
-     *
-     * @return ServerRequestInterface
-     *
-     * @throws \InvalidArgumentException
-     *  If no valid method or URI can be determined.
-     */
-    public function createServerRequestFromArray(array $server);
+    public function createServerRequest($method, $uri, $server = []);
```

Comment:
@shadowhand What we are discussing is about having a way to create a serverRequest instance passing the uri and method (like `RequestInterface`) but also the server params (the only value that we cannot set after initialization).
This standard way to create instances should NOT contain logic. I mean:

* The HTTP_* values in the server params are not converted to http headers
* The SERVER_NAME, REQUEST_URI, QUERY_STRING and other involved values used to build the final uri are not processed
* The Cookie header is not used to Cookies param
* The stream body is not parsed and converted to uploadedFiles

This also prevent to be worried about what happens if some of these values are missing in the server array such `$factory->createServerRequestFromArray([])`. What errors/exceptions should be used?

In addition to this, we could consider to create a new `ServerRequestFromGlobalsFactoryInterface` (or something similar. I think it's bad idea, but anyway), but this should be in a different interface, that could be implemented or not (currently both factories are in the same interface). In fact, the `ServerRequestFromGlobalsFactoryInterface` could use internally `ServerRequestFactoryInterface` to create the server request instance, and only worry about the logic needed to get the final values.